pkg/collector/corechecks/cluster/orchestrator/collector_bundle.go

```diff
@@ -72,25 +75,65 @@ func (cb *CollectorBundle) prepare() {
 
 // prepareCollectors initializes the bundle collector list.
 func (cb *CollectorBundle) prepareCollectors() {
-	// No collector configured in the check configuration.
-	// Use the list of stable collectors as the default.
-	if len(cb.check.instance.Collectors) == 0 {
-		cb.collectors = cb.inventory.StableCollectors()
+	// Custom collector list provided in the check configuration.
+	if len(cb.check.instance.Collectors) > 0 {
```

Comment:
we can't because there's a return statement after the loop meant to make sure we don't continue with autodiscovery or the default config if at least 1 collector has been configured from the check config

Reply:
hmm I see. Then maybe something like this for clarity?
```
	if len(cb.check.instance.Collectors) == 0 {
return
}

for ....
```
saves us the indentation as well

Reply:
Oh yeah that makes a lot of sense, thanks for elaborating! I read too quick. 
Maybe something like encapsulating the strategies and only running some of them if some cases are true? To me it feels relying on specific returns at specific places can lead to error cases during refactoring? 

Maybe something like this helps with readability? There might be another way to make it "safer"? Maybe a test? I am not 100% sure here tbh.

```
if userConfiguredCollectors := retrieveUserConfiguredCollectors(...); if Len(userConfiguredCollectors) >0 {
   return
}

if DiscoverCollectors(){return}

handleDefaultCase() // or just 	cb.collectors = cb.inventory.StableCollectors()
```

Reply:
I tried to comment those statements to describe the logic but it looks like it's not enough.
I'll try to come up with something that is maybe more explicit.

```golang
	if ok := cb.setCollectorsFromCheckConfig(...); ok {
		return
	}
	if ok := cb.setCollectorsFromDiscovery(...); ok {
		return
	}
	cb.setCollectorsFromInventory()
```

Reply:
yes the comments are great but I do believe your provided example is even more explicit and clear imo 🙇

Reply:
I updated the code with this approach